holoviews/ipython/magics.py

```diff
@@ -200,27 +194,15 @@ def list_backends():
             backends += ['%s:%s' % (backend, mode) for mode in modes]
         return backends
 
-    def get_backend_formats(inbuilt_formats=inbuilt_formats):
-        """
-        Build a dictionary of allowed figure and holomap formats
-        across all the available backends and modes.
-        """
-        formats = {}
-        for renderer in Store.renderers.values():
-            mode_formats = renderer.mode_formats
-            assert sorted(mode_formats['fig'].keys()) == sorted(mode_formats['holomap'].keys())
-            for mode in mode_formats['fig'].keys():
-                key = '%s:%s' % (renderer.backend, mode) if mode != 'default' else renderer.backend
-                formats.update({key: (mode_formats['fig'][mode],
-                                      inbuilt_formats + mode_formats['holomap'][mode])})
-        return formats
-
-
+    def list_formats(format_type):
+        backend = Store.current_backend
+        renderer = Store.renderers[backend]
+        return renderer.params(format_type).objects
 
     # Lists: strict options, Set: suggested options, Tuple: numeric bounds.
-    allowed = {'backend'     : list_backends() + ['nbagg', 'd3'],
-               'fig'         : ['svg', 'png', 'repr', 'pdf', 'html'],
-               'holomap'     : inbuilt_formats,
+    allowed = {'backend'     : list_backends(),
+               'fig'         : list_formats('fig'),
+               'holomap'     : list_formats('holomap'),
```

Comment:
If ``list_formats`` is updated as suggested above, the default mode (on initialization of the magic) could be made explicit. e.g ``list_formats('fig', 'mpl:default')`` (or similar).